### PR TITLE
Add optin function to IdentityStore

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
add function that POSTs to 'optin/' endpoint

- [x] add function
- [x] add test(s)

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@imsickofmaps please review

File: lib/identity_store.js [hunk 1]
```diff
@@ -180,6 +180,29 @@ IdentityStore.prototype = {
...
+    :param object identity:
+      Gets passed through directly to service. Consult service docs/protocol.
+      An example of identity object:
+        {
+            "identity": "uuid",
+            "address_type": "msisdn",
+            "address": "+27845000000"
+        }       
+    */
+    optin: function(identity) {
```
Comment (imsickofmaps) [inline lib/identity_store.js]: can we make the 3 params obvious to the user by splitting them out and using a `payload` builder like the other functions?

**Comment by NicolaasDieBaas** [comment]
@imsickofmaps 2nd pass review please

**Comment by imsickofmaps** [comment]
@NicolaasDieBaas sorry -this is too complex - please just simplify as per https://westerncapelabs.slack.com/archives/prk-momconnect-za/p1472217861000042

**Comment by imsickofmaps** [comment]
@NicolaasDieBaas ROFL. I really am sucking at communication on this one! 😆 

`IdentityStore.optin(identity, address_type, address)` is what I had in mind for the function interface

don't forget to clean up the docstrings.

**Comment by NicolaasDieBaas** [comment]
@imsickofmaps a final check please.  hopefully we're on the same page now ;)  if a 👍 from you I'll resolve conflicts with develop and land. thanks

**Comment by imsickofmaps** [comment]
:shipit: 👍

**Pull request merged**